kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/ProxyChannelStateMachine.java

```diff
@@ -472,7 +474,11 @@ void onClientException(@Nullable Throwable cause, boolean tlsEnabled) {
      * @return Return the session ID which connects a frontend channel with a backend channel
      */
     public String sessionId() {
-        return Objects.requireNonNull(sessionId);
+        return kafkaSession.sessionId();
```

Comment:
Since `sessionId` is in principal nullable, we should keep the `requireNonNull`.